### DISCUSSION TITLE
Setting minimum and maximum window size

### DISCUSSION THREAD

**Discussion by Midhil457** [4ian/GDevelop#3631] [Enhancement ideas (and their technical discussions)]
## Description

It's really weird that there is no option to set minimum and maximum window size. Because of that, you can do things like this
![window](https://user-images.githubusercontent.com/73597906/134797472-4177c3b6-a219-4158-b9cf-65c228c60d53.png)

And, according to the geame implementation, it could break the game, spoling everything.
And making the UI responisive and make the game work in all these sizes is also hard.
A bonus would be, there could be a `keep size ratio`, which will keep the size ratio of the window when scaling

## Solution suggested

**Describe the solution**
An option in the game properties and action to change them during the venet is executed would be great

## Alternatives considered

Set the miniumum window size in the engine itself

**Comment by Silver-Streak** [comment]
This basically exists today via events.

Set up an event that checks for screen width and if it is below a certain amount, set the width to the minimum. Same for height.

**Comment by Midhil457** [comment]
I tried it right now, but I wasn't able to get it working, it might be my mistake

Also, I remember the `Change window size` action having a warning that the action shouldn't be called all the time, but it doesn't seem to exist now

but, I think it could be done via events, but I wasn't able to get it working

**Comment by Silver-Streak** [comment]
![image](https://user-images.githubusercontent.com/24237149/135079060-6398254a-4f71-4ce9-957c-8a94086418cf.png)

Just did a quick test, this works for me.

**Comment by Midhil457** [comment]
Tried it and it works. But a issue I found is that it doesn't stop the user from resizing to a value below the minimum value. I change the window size ones the user stops resizing

**Comment by Silver-Streak** [comment]
While I don't think that's a huge issue (it forces it resized if the user keeps trying to resize smaller) you could always disable resizing as part of the action, then re-enable it after a timer.

![image](https://user-images.githubusercontent.com/24237149/135272442-145c2f0f-d4f3-4d06-b6bf-f2e2dfea647d.png)
